src/plots/cartesian/graph_interact.js

```diff
@@ -778,8 +778,9 @@ function createHoverText(hoverData, opts) {
     var i, traceHoverinfo;
     for (i = 0; i < hoverData.length; i++) {
         traceHoverinfo = hoverData[i].trace.hoverinfo;
-        if (traceHoverinfo.indexOf('all')===-1 &&
-                traceHoverinfo.indexOf(hovermode)===-1) {
+        var parts = traceHoverinfo.split('+');
```

Comment:
@alexcjohnson this has me thinking that perhaps `flaglist` attributes should be stored in `fullData` as arrays instead of strings, to make comparison more seamless. Moreover, maybe the `all` flag should be expected to e.g. `x+y+text+name` in the default step.

Alternatively, we could cook a nice general lib function `isInFlagList` that would take care of the logic on demand.